### PR TITLE
Add Benchmarking

### PR DESCRIPTION
**Pull request by ericmann**
Track document sizes and operational performance across several records of size 1 byte through 4MB.

Also use a more performant mapping to convert from character buffers to a string.

### CODE REVIEW

**Comment by ericmann** [comment]
> just running each test once won't give very useful results

I agree. Unfortunately, JS testing doesn't allow for easy segmentation of test suites, so building in a capability of running each test multiple times makes this one test a painful bottleneck for running the rest of our integration tests.

For our immediate use wrt profiling, I ran the suite in isolation 10 times and averaged the results so we'd have something more useful and less susceptible to jitter: https://docs.google.com/spreadsheets/d/1oJYzSaSeeWc5M8kTN-PZiEx9wZaX10zP_84tG4ZlJXA/edit?usp=sharing

> Does a library exist for Node that can do similar?

Not that I've found. The few that are promising were either too limited for what we wanted, measured the wrong things (i.e. direct Node internals rather than overhead from our functions), or were out of date and wouldn't work with current frameworks.

File: lib/__tests__/profile.test.js [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+ */
+function printSizes() {
+  let dats = [data1B, data500B, data250KB, data500KB, data1MB, data2MB, data4MB]
+  let recs = [rec1B, rec500B, rec250KB, rec500KB, rec1MB, rec2MB, rec4MB]
+  let sigs = [signed1B, signed500B, signed250KB, signed500KB, signed1MB, signed2MB, signed4MB]
+
+  for (let row of zip(dats, zip(recs, sigs))) {
+    let data = row[0]
+    let encrypted = row[1][0]
+    let signed = row[1][1]
```
Comment (gstro) [inline lib/__tests__/profile.test.js]: Doesn't look like you're using `signed` here?

File: lib/__tests__/profile.test.js [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    signingKeys.publicKey,
+    signingKeys.privateKey
+  )
+
+  client = new Client(config)
+
+  // Create an EAK for crypto
+  let ak = await Crypto.randomKey()
+  let encryptedAk = await Crypto.encryptAk(cryptoKeys.privateKey, ak, cryptoKeys.publicKey)
+  eak = new EAKInfo(encryptedAk, clientId, cryptoKeys.publicKey, clientId, signingKeys.publicKey)
```
Comment (gstro) [inline lib/__tests__/profile.test.js]: Nice job keeping this from hitting the server.

File: lib/crypto.js [hunk 3]
```diff
@@ -108,7 +108,7 @@ export default class Crypto {
         let dk = sodium.crypto_secretbox_open_easy(edk, edkN, accessKey)
         let field = sodium.crypto_secretbox_open_easy(ef, efN, dk)
 
-        decrypted.data[key] = String.fromCharCode.apply(null, field)
+        decrypted.data[key] = Buffer.from(field).toString('utf8')
```
Comment (gstro) [inline lib/crypto.js]: Nice job grabbing the low-hanging fruit.

File: lib/__tests__/profile.test.js [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+      verify = Date.now() - pause
+      sign = pause - start
+    } while (false)
+
+    console.log(
+      util.format(
+        'Size 500B; Encrypt %dms; Decrypt %dms; Sign %dms; Verify %dms;',
+        enc, dec, sign, verify
+      )
+    )
```
Comment (gstro) [inline lib/__tests__/profile.test.js]: Seems like you could refactor to a single function used by each test, passing in the `dataN` and the string describing the size (e.g. "500 B").

**Pull request merged**